File: Platform.App/node_template/api/queryController.js
var MapBuilder = require("../mapper/builder.js");
var facade = new MapBuilder().build();
var mapperIndex = facade.index;
var mapper = facade.transform;

var domain = require("../model/domain.js")
var ValidityPolicy = require("../model/validityPolicy");


class QueryController{
    
    //Faz a busca baseda num mapa, num app e numa entidade de um subdominio
    getEntityByAppId(req,res,next){
        var appId = req.params.appId;
        var mappedEntity = req.params.entity;
        var entity = mapperIndex.getModelName(appId,mappedEntity);
        var projection = mapperIndex.getProjection(appId,mappedEntity)[mappedEntity];
        projection.include = mapper.getIncludes(appId,mappedEntity,domain);
        if (projection.include.length == 0){
            delete projection.include;
        }
        projection.where = mapper.getFilters(appId,mappedEntity,req);
        if (Object.keys(projection.where).length === 0){
            delete projection.where;
        }
        var vigencia = new ValidityPolicy(appId,mappedEntity,entity);
        vigencia.query(projection, (result)=>{
            res.send(result);
            next();
        }, (e)=>{
            console.log(e);
            res.send("error");
            next();
        });        
    }    
}

module.exports = QueryController;

File: Platform.App/python-template/app/controllers/query_controller.py
""" Query Controller """


class QueryController:
    def __init__(self, app_id, entity, query, mapper, query_service):
        self.app_id = app_id
        self.mapped_entity = entity
        self.page = int(query.pop('page', 0))
        self.page_size = int(query.pop('page_size', 0))
        self.query_string = query
        self.mapper = mapper
        self.query_service = query_service

    def query(self):
        """ query data on domain """
        entity = self.mapper.index.get_model_name(
            self.app_id, self.mapped_entity)
        projection = self.mapper.index.get_projection(self.app_id)

        if not self.mapped_entity in projection:
            raise Exception(f"{self.mapped_entity} not found in map")

        projection = projection[self.mapped_entity]
        projection['where'] = self.mapper.transform.get_filters(self.app_id, self.mapped_entity, self.query_string)

        if not list(projection["where"].keys()):
            projection.pop('where', None)


        result = self.query_service.filter(self.app_id, self.mapped_entity, entity, projection, self.page, self.page_size)
        #result = self.mapper.transform.apply_runtime_fields(self.app_id,self.mapped_entity,result)
        return result
